modules/salt/pages/salt-rate-limiting.adoc

```diff
@@ -0,0 +1,54 @@
+[[salt.rate.limiting]]
+= Rate Limiting
+
+Salt is able to run many commands in parallel on a large number of clients.
+This can potentially create large amounts of load on your infrastructure.
+You can use these rate-limiting features to control the load in your environment.
+
+These features are all configured in the [filename]``/etc/rhn/rhn.conf`` configuration file.
+
+
+
+== Batch Size
+
+Actions are processed in batches.
+This parameter sets the maximum number of clients that can execute a single action in each batch.
+
+Choosing a lower number can prevent system thrashing, but might reduce overall performance as parallelization is reduced.
+
+Adjust the [systemitem]``java.salt_batch_size`` parameter.
+Defaults to 200.
+
+
+
+== Batch Delay
+
+This parameter sets the amount of time, in seconds, to wait after a batch is processed before beginning to process the next batch.
```

Comment:
> 3. that as soon as any minion is done, another minion immediately* starts (**unlike** in batching)
>    - corner case: more than one minion is done precisely at the same time. An equal number of minions start

*: this parameter sets a certain delay after one or more minions are done and the time new command(s) are issued to an equal number of minions.

Why would one want to add such a delay, instead of starting immediately? Doesn't that harm performance?

It might not, as starting immediately would probably mean (in the example above) that the command is initially sent to 20 minions all at once, then 1 is done and 1 is issued, then 1 is done and 1 is issued, then 1 is done and 1 is issued... 80 times. For a grand total of 81 issued commands.

If we set a batch delay of, say, 5 seconds, chances are that several minions might complete before the next command is issued, thus giving Salt the opportunity to collect the results and issue less commands, For example I can imagine a first batch of 20, then we get 10 minion results before the 5 seconds elapse so we issue one command to 10 target minions, then we get 12, then maybe 8... The grand total could be 8-10 commands, much less than the case above.

This is good because of how Salt implements broadcasting to minions (many commands each targeting one minion are much less efficient than a few targeting big groups).